### PR TITLE
MassInstallAction: Install the GitHubActions workflow on this repository

### PR DESCRIPTION
**Pull request by fchorney**
This pull request sets up the GitHubActions workflow on this repository. cc: @fchorney

### CODE REVIEW

**Review by oxinabox — APPROVED** [review]

**Comment by fchorney** [comment]
I think the documentation job isn't picking up that Compat has an update?

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      - run: |
+          git config --global user.name name
+          git config --global user.email email
+          git config --global github.user username
+      - run: |
+          julia --project=docs -e '
+            using Pkg;
+            Pkg.develop(PackageSpec(path=pwd()));
+            Pkg.instantiate();
+            include("docs/make.jl");'
```
Comment (omus) [inline .github/workflows/CI.yml]:
```suggestion
          julia --project=docs -e '
            using Pkg
            Pkg.develop(PackageSpec(path=pwd()))
            Pkg.instantiate()
            include("docs/make.jl")'
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: '1'
+      - run: |
+          git config --global user.name name
+          git config --global user.email email
+          git config --global github.user username
```
Comment (omus) [inline .github/workflows/CI.yml]:
```suggestion
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - uses: julia-actions/julia-buildpkg@latest
+      - run: |
+          git config --global user.name Tester
+          git config --global user.email te@st.er
```
Comment (omus) [inline .github/workflows/CI.yml]:
```suggestion
```

**Review by omus — COMMENTED** [review]
These changes aren't necessary but I thought I would point them out

**Comment by fchorney** [comment]
> These changes aren't necessary but I thought I would point them out

Interesting, so none of those are required?

**Comment by omus** [comment]
> > These changes aren't necessary but I thought I would point them out
> 
> Interesting, so none of those are required?

They [aren't used by PkgTemplates.jl](https://github.com/invenia/PkgTemplates.jl/blob/master/templates/github/workflows/CI.yml) so I'm fairly sure they aren't required

**Pull request merged**